go/store/types/map_leaf_sequence.go

```diff
@@ -350,3 +367,216 @@ func (ml mapLeafSequence) getValue(idx int) (Value, error) {
+	return nil, nil
+}
+
+func (mes mapEntrySequence) getItem(idx int) (sequenceItem, error) {
+	return mes.entries[idx], nil
+}
+
+func (mes mapEntrySequence) isLeaf() bool {
+	return true
+}
+
+func (mes mapEntrySequence) Kind() NomsKind {
+	return MapKind
+}
+
+func (mes mapEntrySequence) Len() uint64 {
+	return uint64(len(mes.entries))
+}
+
+func (mes mapEntrySequence) numLeaves() uint64 {
+	return uint64(len(mes.entries))
+}
+
+func (mes mapEntrySequence) seqLen() int {
+	return len(mes.entries)
+}
+
+func (mes mapEntrySequence) treeLevel() uint64 {
+	return 0
+}
+
+func (mes mapEntrySequence) valueReadWriter() ValueReadWriter {
+	return mes.vrw
+}
+
+func (mes mapEntrySequence) valuesSlice(from, to uint64) ([]Value, error) {
+	if l := mes.Len(); to > l {
+		to = l
+	}
+
+	numPairs := (to - from)
+	numTuples := numPairs * 2 // 1 key, 1 value interleaved
+	dest := make([]Value, numTuples)
+
+	dest = dest[:numTuples]
+	for i := uint64(0); i < numPairs; i++ {
+		entry := mes.entries[i]
+		dest[i*2] = entry.key
+		dest[i*2+1] = entry.value
+	}
+	return dest, nil
+}
+
+func (mes mapEntrySequence) kvTuples(from, to uint64, dest []Tuple) ([]Tuple, error) {
+	if l := mes.Len(); to > l {
+		to = l
+	}
+
+	numPairs := (to - from)
+	numTuples := numPairs * 2 // 1 key, 1 value interleaved
+
+	if uint64(cap(dest)) < numTuples {
+		dest = make([]Tuple, numTuples)
+	}
+
+	dest = dest[:numTuples]
+	for i := uint64(0); i < numPairs; i++ {
+		entry := mes.entries[i]
+
+		keyTuple, ok := entry.key.(Tuple)
+
+		if !ok {
+			return nil, ErrNotAMapOfTuples
+		}
+
+		valTuple, ok := entry.value.(Tuple)
+
+		if !ok {
+			return nil, ErrNotAMapOfTuples
+		}
+
+		dest[i*2] = keyTuple
+		dest[i*2+1] = valTuple
+	}
+	return dest, nil
+}
+
+func (mes mapEntrySequence) getCompareFn(other sequence) compareFn {
+	if otherCMLS, ok := other.(mapEntrySequence); ok {
+		return func(idx, otherIdx int) (bool, error) {
+			entry1 := mes.entries[idx]
+			entry2 := otherCMLS.entries[otherIdx]
+
+			if !entry1.key.Equals(entry2.key) {
+				return false, nil
+			}
+
+			return entry1.value.Equals(entry2.value), nil
+		}
+	} else if ml, ok := other.(mapLeafSequence); ok {
+		dec1 := ml.decoder()
+		return func(mesIdx, mlsIdx int) (bool, error) {
+			entry := mes.entries[mesIdx]
+			dec1.offset = uint32(ml.getItemOffset(mlsIdx))
+			mlk, err := dec1.readValue(ml.format())
+
+			if err != nil {
+				return false, err
+			}
+
+			if !entry.key.Equals(mlk) {
+				return false, nil
+			}
+
+			mlv, err := dec1.readValue(ml.format())
+
+			if err != nil {
+				return false, err
+			}
+
+			return entry.value.Equals(mlv), nil
+		}
+	}
+
+	panic("unsupported")
+}
+
+func (mes mapEntrySequence) typeOf() (*Type, error) {
+	panic("implement me")
+}
+
+func (mes mapEntrySequence) WalkRefs(nbf *NomsBinFormat, cb RefCallback) error {
+	panic("implement me")
+}
+
+func (mes mapEntrySequence) writeTo(writer nomsWriter, format *NomsBinFormat) error {
+	panic("implement me")
```

Comment:
AKAIK we're not using `getCompositeChildSequence` on the write path anywhere. I believe we'll panic here if we try to.

Reply:
yeah, Goland auto filled the implementation for me.  Changed the panic methods as I'm not planning on implementing.